### PR TITLE
Add auditor for undefined scheme in post_logout_redirect_uris

### PR DESCRIPTION
**Pull request by dasniko**
# Summary

- Adds `ClientHasUndefinedBaseDomainAndSchemaInPostLogoutRedirectUri`, an Info-level `ClientAuditor` that checks whether each entry in a client's `post_logout_redirect_uris` resolves to a fully qualified URI with a defined scheme.
- Mirrors the existing `ClientHasUndefinedBaseDomainAndSchema` auditor for regular redirect URIs, applying the same resolution logic (prepend `rootUrl` for relative URIs).
- Keycloak's `+` special value ("inherit from redirect URIs") is explicitly skipped.
- Adds `get_post_logout_redirect_uris()` and `get_resolved_post_logout_redirect_uris()` accessors to the Client data model, handling Keycloak's ##-separated attribute format (`attributes["post.logout.redirect.uris"]`).

# Motivation

According to the https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RedirectionAfterLogout, the OpenID Provider redirects the user back to the Relying Party after logout using a URI from the client's registered `post_logout_redirect_uris`. The https://openid.net/specs/openid-connect-rpinitiated-1_0.html#ClientMetadata states these URIs **should** use the HTTPS scheme.

A `post_logout_redirect_uri` with an undefined or unresolvable scheme (e.g. `//example.com/logout` or `example.com/logout`) will be silently unusable — the OP cannot redirect to it, and the misconfiguration is not surfaced by Keycloak at configuration time.

# Test plan

- `poetry run pytest tests/auditors/client/test_client_has_undefined_base_domain_and_schema_in_post_logout_redirect_uri.py` — 18 unit tests covering `should_consider_client`, scheme detection, all edge cases including the `+` special value and `##`-separated multi-value parsing, and absent attribute handling
- `poetry run ruff check . --fix && poetry run ruff format .` — no issues

# AI disclosure

The code of this PR was created with support of Claude Code, yes.
As I don't have the proper Python knowledge, but want to contribute to this awesome project, I decided to give it a try. I'd be happy and it would be awesome if you merge this into main, do some adjustments and/or leave some comments about what to improve. I'll do my very best. ❤️

### CODE REVIEW

File: docs/auditors/clients.md [hunk 1]
```diff
@@ -96,6 +96,15 @@ This auditor identifies clients where the redirect URI, in combination with the
 This often indicates that a fully qualified domain name, including the scheme (e.g., 'https://example.com/login'), is not defined for either the client root URL or the redirect URIs.
 To address this issue, clients should specify clear redirect URIs with proper schemes to enhance security.
 
+## ClientHasUndefinedBaseDomainAndSchemaInPostLogoutRedirectUri
+
+This auditor checks if OIDC clients have undefined or insufficiently specified `post_logout_redirect_uri` schemes.
+After an RP-initiated logout, the OpenID Provider redirects the user back to the Relying Party using a URI from the client's registered `post_logout_redirect_uris`.
+According to the [OIDC RP-Initiated Logout specification](https://openid.net/specs/openid-connect-rpinitiated-1_0.html#ClientMetadata), these URIs should use the HTTPS scheme.
+This auditor identifies clients where a `post_logout_redirect_uri`, in combination with the client's root URL, does not resolve to a fully qualified address with a defined scheme.
+An unresolvable URI cannot be used by the OpenID Provider for post-logout redirection and will silently fail.
```
Comment (malexmave) [inline docs/auditors/clients.md]: Is this just "according to the spec", or do you know from (painful) experience that Keycloak behaves like this? ;)

**Review by malexmave — COMMENTED** [review]
Hi Niko. Same answer here - thanks for the contribution and nice words, and a few questions inline ;).

**Review by malexmave — APPROVED** [review]
Same here - looks good to me, @twwd can take a look or merge directly :).
